public/app/features/inspector/QueryInspector.tsx

```diff
@@ -47,9 +49,21 @@ export class QueryInspector extends PureComponent<Props, State> {
   }
 
   componentDidMount() {
+    const { instanceId } = this.props;
     this.subs.add(
       backendSrv.getInspectorStream().subscribe({
-        next: (response) => this.onDataSourceResponse(response),
+        next: (response) => {
+          let update = true;
+          if (instanceId && response?.config?.url) {
```

Comment:
this "should" make this work even is the `instanceId` prop changes
```suggestion
          if (this.props.instanceId && response?.config?.url) {
```